src/PuuClocks.API/cmd/main.go

```diff
@@ -32,6 +38,51 @@ func main() {
 		})
 	})
 
+	lobbyManager := sockets.NewLobbyManager()
+	r.POST("/create-lobby", func(c *gin.Context) {
+		lobby := lobbyManager.CreateLobby()
+		c.JSON(http.StatusOK, gin.H{
+			"lobbyID": lobby.GetID(),
+		})
+	})
+
+	r.GET("/join-lobby/:id", func(c *gin.Context) {
+		var conn *websocket.Conn
+		var parsedID uuid.UUID
+		conn, err = sockets.Upgrader.Upgrade(c.Writer, c.Request, nil)
+		if err != nil {
+			log.Log.Errorln(err)
+			return
+		}
+		defer conn.Close()
+		id := c.Param("id")
+
+		parsedID, err = uuid.Parse(id)
+		if err != nil {
+			if err = conn.WriteJSON(map[string]string{
+				"message": "User not passed lobby UUID",
+			}); err != nil {
+				log.Log.Errorln(err)
+			}
+		} else {
+			l := lobbyManager.FindLobby(parsedID)
+			if l == nil {
+				if err = conn.WriteJSON(map[string]string{
+					"message": "Lobby not found",
+				}); err != nil {
+					log.Log.Errorln(err)
+				}
+			} else {
+				sockets.NewClient(conn, l)
```

Comment:
Ale jak User poda dobre id to nie dodawaj go od razu do socketa tylko zwróć mu, że id jest poprawne, a potem client ma się bezpośrednio do tego socketa podłaczyć

Reply:
albo może nawet zrobić z tego POST'a i w body wrzucać hasło jeśli lobby ma hasło, jeśli jest poprawne to zwraca wtedy adres websocketa do którego client ma się podłączyć